### PR TITLE
feat(test-suite): add operator edge-case e2e tests

### PR DESCRIPTION
**Pull request by PanGan21**
### Summary

- `shl`/`shr`/`rotl`/`rotr` at amount `0`, `W-1`, `W`, `255` — `euint8`..`euint256`, scalar and encrypted RHS
- `div`/`rem` identities, boundaries, zero dividend, `DivisionByZero()` revert
- `add`/`sub`/`mul`/`neg`/`not` wrapping on over/underflow
- narrowing casts truncating to the target width
- `OVERSHIFT_RETURNS_ZERO` (`shiftSemantics.ts`): flip to `true` with the tfhe-rs >= 1.7.0 bump; single point of change for the new overshift behaviour

### Run

```sh
./fhevm-cli test operators --grep "edge cases" --verbose
```

Related: https://github.com/zama-ai/fhevm-internal/issues/1572

### CODE REVIEW

**Comment by PanGan21** [comment]
E2E test run for edge cases: https://github.com/zama-ai/fhevm/actions/runs/33063841864

File: test-suite/e2e/test/fhevmOperations/manual.ts [hunk 1]
```diff
@@ -1838,3 +1864,156 @@ describe('FHEVM manual operations', function () {
     assert.equal(res.clearValues[handle], ADDR_A);
   });
 });
+
+// Top-level: these tests use their own fixture, so they must not pay for the
+// FHEVMManualTestSuite deploy in the parent beforeEach (shared signer, nonce pressure).
+describe('FHEVM manual operations - shift, rotate, div and rem edge cases', function () {
+  beforeEach(async function () {
```
Comment (antoniupop) [inline test-suite/e2e/test/fhevmOperations/manual.ts]: Stateless afaict, so maybe just before so we don't do it every time?

**Comment by antoniupop** [comment]
Another concern is total runtime - is this be parallelised? Would it need splitting in separate files?

File: test-suite/e2e/test/fhevmOperations/manual.ts (hunk 1 above)
Comment (PanGan21) [reply test-suite/e2e/test/fhevmOperations/manual.ts]: Yes you are right it is stateless. I switched to `before` in: https://github.com/zama-ai/fhevm/pull/3638/commits/83ed280935f4071f35a543c9434bf5f1d751f73c

**Comment by PanGan21** [comment]
> Another concern is total runtime - is this be parallelised? Would it need splitting in separate files?

I have splitted the tests into four files grouped by operators. I moved them out of `manual.ts` which allows mocha for better parallelization using: `./fhevm-cli test operators --parallel --grep "edge cases"`
New e2e test run with the updated code in progress: https://github.com/zama-ai/fhevm/actions/runs/33071851216

**Review by antoniupop — DISMISSED** [review]

**Review by antoniupop — APPROVED** [review]
